src/Symfony/Component/Notifier/Bridge/Smsbox/SmsboxOptions.php

```diff
@@ -0,0 +1,381 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Smsbox;
+
+use Symfony\Component\Intl\Countries;
+use Symfony\Component\Notifier\Exception\InvalidArgumentException;
+use Symfony\Component\Notifier\Message\MessageOptionsInterface;
+
+/**
+ * @author Alan Zarli <azarli@smsbox.fr>
+ * @author Farid Touil <ftouil@smsbox.fr>
+ */
+final class SmsboxOptions implements MessageOptionsInterface
+{
+    public const MESSAGE_MODE_STANDARD = 'Standard';
+    public const MESSAGE_MODE_EXPERT = 'Expert';
+    public const MESSAGE_MODE_RESPONSE = 'Reponse';
+
+    public const MESSAGE_STRATEGY_PRIVATE = 1;
+    public const MESSAGE_STRATEGY_NOTIFICATION = 2;
+    public const MESSAGE_STRATEGY_NOT_MARKETING_GROUP = 3;
+    public const MESSAGE_STRATEGY_MARKETING = 4;
+
+    public const MESSAGE_CODING_DEFAULT = 'default';
+    public const MESSAGE_CODING_UNICODE = 'unicode';
+    public const MESSAGE_CODING_AUTO = 'auto';
+
+    public const MESSAGE_CHARSET_ISO_1 = 'iso-8859-1';
+    public const MESSAGE_CHARSET_ISO_15 = 'iso-8859-15';
+    public const MESSAGE_CHARSET_UTF8 = 'utf-8';
+
+    public const MESSAGE_DAYS_MONDAY = 1;
+    public const MESSAGE_DAYS_TUESDAY = 2;
+    public const MESSAGE_DAYS_WEDNESDAY = 3;
+    public const MESSAGE_DAYS_THURSDAY = 4;
+    public const MESSAGE_DAYS_FRIDAY = 5;
+    public const MESSAGE_DAYS_SATURDAY = 6;
+    public const MESSAGE_DAYS_SUNDAY = 7;
+
+    public const MESSAGE_UDH_6_OCTETS = 1;
+    public const MESSAGE_UDH_7_OCTETS = 2;
+    public const MESSAGE_UDH_DISABLED_CONCAT = 0;
+
+    private array $options;
+
+    public function __construct(array $options = [])
+    {
+        $this->options = [];
+    }
+
+    /**
+     * @return null
+     */
+    public function getRecipientId(): ?string
+    {
+        return null;
+    }
+
+    public function mode(string $mode)
+    {
+        $this->options['mode'] = self::validMode($mode);
+
+        return $this;
+    }
+
+    public function strategy(int $strategy)
+    {
+        $this->options['strategy'] = self::validStrategy($strategy);
+
+        return $this;
+    }
+
+    public function date(string $date)
+    {
+        $this->options['date'] = self::validDate($date);
```

Comment:
Can't we use a `DateTime` here and format it internally to?

Reply:
Hello,

If the integration is based on the "datetime()" option, the original time zone will be used to allow transparent conversion to the "Europe/Paris" time zone.
 
On the other hand, historically our API allows you to pass the "date() & hour()" options, but these two parameters are based directly on the "Europe/Paris" time zone without conversion.
We expect a date in this format "dd/mm/yyyy", i.e. a string, as documented in our API.
And because we have many users who already integrate these options in this form, we want to keep them and add more flexibility with the "datetime()" option.
Thank you.

Reply:
we should rename `valid*()` to `validate*()`

Reply:
fixed